src/ATen/native/xpu/sycl/WeightInt4PackKernel.cpp

```diff
@@ -9,19 +9,22 @@ namespace at::native::xpu {
 struct WeightToInt4PackKernelFunctor {
   void operator()(sycl::item<1> item) const {
     auto idx = item.get_linear_id();
-    int out_y = idx / N_;
-    int out_x = idx % N_;
-    int in_y = out_x;
-    int in_x = out_y * 4;
     int K_div_2 = K_ / 2;
+    int K_div_8 = K_ / 8;
```

Comment:
although might never happened, but could you handle ```K_``` not divided by 8?

Reply:
good point, let me add an assert